### ISSUE TITLE
Multiple "actions" per workflow

### ISSUE THREAD

**Issue by space-nuko** [comfyanonymous/ComfyUI#528]
Currently I think ComfyUI supports only one group of input/output per graph. I want to be able to run multiple different scenarios per workflow. For example there's a preview image node, I'd like to be able to press a button an get a quick sample of the current prompt. Then a separate button triggers the longer image generation at full resolution. And that's without manually disconnecting any nodes in the litegraph instance

Another scenario is manually selecting an image out of a batch and running only an upscale part of the highres workflow separate from the two-step process. First there's a way to choose the image you want, which sends it to the latent node, then another button to run step 2. For that case those two parts of the graph are totally disconnected from each other, or alternatively the image output is connected to the highres input somehow but execution doesn't proceed until you press a second button for that part of the workflow

I've also noticed, some people want a global seed node across all sampler nodes. So for this case it's a single number node that copies its value into the samplers it's connected to when a button is pressed

Some programming environments like Max/MSP solve this with "bangs" and input ports. It lets you send a string message like `set index 1 foo` into a special input port that interprets the command and sets the value of the widget. Can help if the action is entirely in the frontend. As for backend maybe the ability to tag nodes as belonging to certain actions and sending an argument to the prompt executor to only run certain nodes in a tag whitelist. In Max all nodes have special properties attached that controls their behavior and appearance, similar to UI designer properties in an IDE

I can see ComfyUI supplying its own set of hooks for lifecycle events like prompt being queued, that generalizes the seed randomization behavior that's a bit hard coded right now. Queuing a prompt would send a "bang" from some "system output node" that you can feed into the seed node, which triggers setting the new seed

Of course this is basically half of what Max/MSP does already but it's worth looking at how they solve these kinds of problems with their own node-based environment

**Comment by bitcrusherrr** [comment]
A way to "pause" on a node completion would be great.

**Comment by space-nuko** [comment]
I don't think it helps when you want more than two actions, what I expect is for example three buttons "Preview" "Batch" and "HR Fix" that run three separate stages of the workflow, preferably without needing to touch the graph that much or disconnecting/reconnecting nodes

**Comment by omar92** [comment]
maybe we can have a sppecial reroute node that have a button which mutes and unmutes it

**Comment by space-nuko** [comment]
There's at least two scenarios I can think of with regards to rerouting

- Flipping a switch to disable/enable an extra step
- Toggling between one of multiple options to choose which wire to use

So I think rerouting should work at the wire layer instead of the node layer. As in you'd specify which wire(s) is the target for toggling with a control

**Comment by donengel72** [comment]
I would love to be able to pause before saving; IE, I send an image to a previewer and then choose to save the file or stop graph execution, so my output doesn’t get cluttered with poor results while trying to get things right.

would be generally great to be able to breakpoint and switch the graph pathways based on user input while the process is running.

**Comment by Tekaiguy** [comment]
I believe [Partial Execution](https://docs.comfy.org/interface/features/partial-execution) solves this request.
